### PR TITLE
Build ocean ice interpolation code w/ CMake and add namelists, etc.

### PR DESCRIPTION
**Pull request by aerorahul**
This PR:
- builds the `ocnicepost.fd` code with cmake
- adds CSV files for ocean and ice, as well as sample fortran namelists for running the executable.

### CODE REVIEW

File: parm/ocnicepost/ice.csv [hunk 1]
```diff
@@ -0,0 +1,8 @@
+! variable name, dimension, grid location, remapping method, vector pair, grid location of vector pair
+  'hi_h', 2 , 'Ct', 'bilinear', '', ''
+  'hs_h', 2 , 'Ct', 'bilinear', '', ''
+'aice_h', 2 , 'Ct', 'bilinear', '', ''
+ 'sst_h', 2 , 'Ct', 'bilinear', '', ''
```
Comment (HenryWinterbottom-NOAA) [inline parm/ocnicepost/ice.csv:5]: Fix indentation.

File: parm/ocnicepost/ice.csv [hunk 2]
```diff
@@ -0,0 +1,8 @@
+! variable name, dimension, grid location, remapping method, vector pair, grid location of vector pair
+  'hi_h', 2 , 'Ct', 'bilinear', '', ''
+  'hs_h', 2 , 'Ct', 'bilinear', '', ''
```
Comment (HenryWinterbottom-NOAA) [inline parm/ocnicepost/ice.csv:3]: Fix indentation.

File: parm/ocnicepost/ice_mx025_to_0p25.nml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+& ocnicepost_nml
+ftype='ice'
+wgtsdir='/scratch1/NCEPDEV/stmp4/Denise.Worthen/CPLD_GRIDGEN/'
```
Comment (HenryRWinterbottom) [inline parm/ocnicepost/ice_mx025_to_0p25.nml:3]: This should not be hardcoded. A template (jinja) should be used as part of the GW to create this file.

File: parm/ocnicepost/ocean_mx025_to_0p25.nml [hunk 4]
```diff
@@ -0,0 +1,11 @@
+& ocnicepost_nml
```
Comment (HenryRWinterbottom) [inline parm/ocnicepost/ocean_mx025_to_0p25.nml:1]: https://github.com/NOAA-EMC/gfs-utils/pull/26/files#r1367325588

**Review by HenryRWinterbottom — CHANGES_REQUESTED** [review]
A few minor cosmetic things and hardcoded paths.

File: parm/ocnicepost/ice_mx025_to_0p25.nml (hunk 3 above)
Comment (aerorahul) [reply parm/ocnicepost/ice_mx025_to_0p25.nml:3]: This is just a sample of what @DeniseWorthen provided on Hera.  
The workflow will use the jinja template to fill it properly.  
Once we write a test for this program, these sample files will be removed.

Comment (aerorahul) [reply parm/ocnicepost/ice_mx025_to_0p25.nml:3]: See `ocnicepost.nml.jinja2` for the template.

**Review by HenryRWinterbottom — APPROVED** [review]
Approved.

File: src/ocnicepost.fd/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,13 @@
+list(APPEND fortran_src
+arrays_mod.F90
+init_mod.F90
+masking_mod.F90
+ocnicepost.F90
+utils_mod.F90
```
Comment (WalterKolczynski-NOAA) [inline src/ocnicepost.fd/CMakeLists.txt]: Not well-versed in cmake. Can this be indented?

Comment (aerorahul) [reply src/ocnicepost.fd/CMakeLists.txt]: done.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**